ArgusCore/src/main/java/com/salesforce/dva/argus/service/schema/ElasticSearchSchemaService.java

```
@@ -723,6 +742,38 @@ private void _upsertScopes(List<ScopeOnlySchemaRecord> records) {
 		}
 	}
 
+	private Response _updateMtsField(List<String> docIds, String index, String type) {
+		Response result= null;
+		if(docIds != null && docIds.size()>0) {
+			String requestUrl = new StringBuilder().append("/")
+					.append(index)
+					.append("/")
+					.append(type)				
+					.append("/")
+					.append("_bulk")
+					.toString();
+			try {
+				String requestBody = _getRequestBodyForMtsFieldUpdate(docIds);
+				result = _esRestClient.performRequest(HttpMethod.POST.getName(), requestUrl, Collections.emptyMap(),
+						new StringEntity(requestBody));
```

we might need the retry_on_conflict request param. If there is version conflict then its possible ES will throw an exception

---

@nkunal  added test cases.. lets create another story for retry logic as the current sprint is over

---

chatted with Kunal, Instead of retry we should change to ignore these failures, as we only want to typically update mts once a day.

---

ok